### PR TITLE
Refactor examples_test.walkJSONFiles

### PR DESCRIPTION
**Pull request by tmrts**
While I was walking through the examples, I found the flow of this function to be confusing, so I decided to make it a bit more clearer. I hope it's better now.

### CODE REVIEW

**Comment by TamerTas** [comment]
There was a missing type declaration in the merged code that was obstructing the builds. Should be okay now.

File: examples/examples_test.go [hunk 1]
```diff
@@ -69,31 +69,26 @@ func validateObject(obj runtime.Object) (errors []error) {
...
-			name = name[:len(name)-len(ext)]
-		}
-		if !(ext == ".json" || ext == ".yaml") {
-			return nil
-		}
-		glog.Infof("Testing %s", path)
-		data, err := ioutil.ReadFile(path)
-		if err != nil {
-			return err
+func walkJSONFiles(inDir string, fn func(name string, path string, data []byte)) error {
```
Comment (roberthbailey) [inline examples/examples_test.go]: This should be fn func(name, path string, data []byte) as it was before. When you have two function arguments of the same type in a row you only need to write the type once.

File: examples/examples_test.go [hunk 2]
```diff
@@ -69,31 +69,26 @@ func validateObject(obj runtime.Object) (errors []error) {
...
+	return filepath.Walk(inDir, func(path string, info os.FileInfo, err error) error {
+		if err == nil {
```
Comment (roberthbailey) [inline examples/examples_test.go]: The old way (handling the error first) is actually the go style; see https://code.google.com/p/go-wiki/wiki/CodeReviewComments#Indent_Error_Flow

File: examples/examples_test.go (hunk 1 above)
Comment (TamerTas) [inline examples/examples_test.go]: Ah, I missed that part. I'm still getting used to reading type after variable names :)

File: examples/examples_test.go (hunk 2 above)
Comment (TamerTas) [inline examples/examples_test.go]: Made the necessary adjustments.

File: examples/examples_test.go [hunk 3]
```diff
@@ -70,30 +70,27 @@ func validateObject(obj runtime.Object) (errors []error) {
...
-			return err
+
+		file := filepath.Base(path)
+		if ext := filepath.Ext(file); ext == ".json" || ext == ".yaml" {
+			glog.Infof("Testing %s", path)
+			data, err := ioutil.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			name := file[:len(file)-len(ext)]
```
Comment (roberthbailey) [inline examples/examples_test.go]: strings.TrimSuffix(ext) here would probably be clearer than the slicing.

**Comment by roberthbailey** [comment]
One more minor nit. Will merge on Monday during merging window.

**Comment by TamerTas** [comment]
@roberthbailey Thanks

**Pull request merged**